clients/command-kafka/src/main/java/org/eclipse/hono/client/command/kafka/KafkaBasedInternalCommandConsumer.java

```diff
@@ -265,10 +302,33 @@ private String getTopicName() {
             LOG.debug("consumer closed");
             Optional.ofNullable(metricsSupport).ifPresent(ms -> ms.unregisterKafkaConsumer(consumer.unwrap()));
         });
-        return CompositeFuture.all(adminClientClosePromise.future(), consumerClosePromise.future())
+
+        return CompositeFuture.all(deleteTopicIfExists(), consumerClosePromise.future())
                 .mapEmpty();
     }
 
+    private Future<Void> deleteTopicIfExists() {
+        if (isTopicCreated.get()) {
+            final String topicName = getTopicName();
+            final Promise<Void> adminClientClosePromise = Promise.promise();
+            LOG.debug("stop: delete topic [{}]", topicName);
+            adminClient.deleteTopics(List.of(topicName))
+                    .all()
+                    .whenComplete((v, ex) -> {
+                        if (ex != null) {
+                            LOG.warn("error deleting topic [{}]", topicName, ex);
+                        }
+                        context.executeBlocking(future -> {
+                            adminClient.close();
+                            future.complete();
+                        }, adminClientClosePromise);
+                    });
+            return adminClientClosePromise.future().onComplete(ar -> LOG.debug("admin client closed"));
+        } else {
+            return Future.succeededFuture();
```

Comment:
In this case the admin client should also be closed.
Closing the admin client could also be moved out of this method entirely, then the `deleteTopicIfExists` method name would also fit better.